url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/StandardBeanInfo.java

```diff
@@ -0,0 +1,1505 @@
+            String indexedTag = (String) table.get(STR_INDEXED);
+
+            if ((normalTag == null) && (indexedTag == null)) {
+                continue;
+            }
+
+            Method get = (Method) table.get(STR_NORMAL + PREFIX_GET);
+            Method set = (Method) table.get(STR_NORMAL + PREFIX_SET);
+            Method indexedGet = (Method) table.get(STR_INDEXED + PREFIX_GET);
+            Method indexedSet = (Method) table.get(STR_INDEXED + PREFIX_SET);
+
+            PropertyDescriptor propertyDesc = null;
+            if (indexedTag == null) {
+                propertyDesc = new PropertyDescriptor(propertyName, get, set);
+            } else {
+                try {
+                    propertyDesc = new IndexedPropertyDescriptor(propertyName,
+                            get, set, indexedGet, indexedSet);
+                } catch (IntrospectionException e) {
+                    // If the getter and the indexGetter is not compatible, try
+                    // getter/setter is null;
+                    propertyDesc = new IndexedPropertyDescriptor(propertyName,
+                            null, null, indexedGet, indexedSet);
+                }
+            }
+            // RI set propretyDescriptor as bound. FIXME
+            // propertyDesc.setBound(true);
+            if (canAddPropertyChangeListener && canRemovePropertyChangeListener) {
+                propertyDesc.setBound(true);
+            } else {
+                propertyDesc.setBound(false);
+            }
+            if (table.get(STR_IS_CONSTRAINED) == Boolean.TRUE) { //$NON-NLS-1$
+                propertyDesc.setConstrained(true);
+            }
+            propertyList.add(propertyDesc);
+        }
+
+        PropertyDescriptor[] theProperties = new PropertyDescriptor[propertyList
+                .size()];
+        propertyList.toArray(theProperties);
+        return theProperties;
+    }
+
+    private boolean isInSuper(MethodDescriptor method,
+            MethodDescriptor[] excludeMethods) {
+        for (MethodDescriptor m : excludeMethods) {
+            if (method.getMethod().equals(m.getMethod())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @SuppressWarnings("nls")
+    private void introspectPropertyListener(Method theMethod) {
+        String methodName = theMethod.getName();
+        Class<?>[] param = theMethod.getParameterTypes();
+        if (param.length != 1) {
+            return;
+        }
+        if (methodName.equals("addPropertyChangeListener")
+                && param[0].equals(PropertyChangeListener.class))
+            canAddPropertyChangeListener = true;
+        if (methodName.equals("removePropertyChangeListener")
+                && param[0].equals(PropertyChangeListener.class))
+            canRemovePropertyChangeListener = true;
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private static void introspectGet(Method theMethod,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 840-851, lines 860-873 into separate methods.